src/GraphManager/GraphRenderer.tsx

```diff
@@ -295,7 +308,7 @@ export const GraphRenderer = (props: GraphRendererProps) => {
     }
   }, [graphDataInfo]);
   useEffect(() => {
-    convertAndSetGraph(setGraph, graphDataFromBackend);
+    onGraphUpdate(props.controllerRef, setGraph, graphDataFromBackend);
```

Comment:
You could should pass `controller` here. And then create an object `{current: controller}` that is passed to `userSearchMatching`, since a [react ref](https://legacy.reactjs.org/docs/refs-and-the-dom.html) is just that an object with a `current` key.